go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -324,6 +440,51 @@ func newSharedGateway(nodeName string, subnets []*net.IPNet, gwNextHops []net.IP
 	// patch-<logical_port_name_of_localnet_port>-to-br-int
 	patchPort := "patch-" + bridgeName + "_" + nodeName + "-to-br-int"
 
+	// add service route to bridge
+	bridgeLink, err := util.LinkSetUp(bridgeName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find shared gw bridge interface: %s", bridgeName)
+	}
+	for _, subnet := range config.Kubernetes.ServiceCIDRs {
+		nextHops, err := util.MatchIPFamily(utilnet.IsIPv6CIDR(subnet), gwNextHops)
+		if err != nil {
+			klog.Warningf("Unable to find valid next hop for Service CIDR: %s, available next hops: %v",
+				subnet, gwNextHops)
+		}
+
+		// TODO(trozet): see if we need to consider upgrade here...there may be an old route pointing to mp0
+		if exists, err := util.LinkRouteExists(bridgeLink, nextHops[0], subnet); err == nil && !exists {
```

Comment:
L455: yes, I think we will need an upgrade path here.. unfortunately..

Reply:
ok will handle removing the old route.